Fix crawl redirect errors

bookwyrm opened the pull request:
Allow two redirect hops to support new WP instance crawl behavior.
Report on errors from too many redirects in CLI for debugging.

bookwyrm committed:
Increase allowed redirect count

This fixes a "Too Many Redirects" error when a fresh site redirects twice to go from `/` to the actual post front page.

bookwyrm committed:
Report on redirects in CLI

So that we can have a place to start debugging when we encounter a "Too Many Redirects" error.

bookwyrm committed:
Changelog update

bookwyrm committed:
Fixes for PHPCS

john-shaffer commented on src/Crawler.php:
```diff
@@ -219,7 +221,14 @@ public function crawlURL( string $url ) : ?ResponseInterface {
+            $response = $this->client->send( $request );
+        } catch ( TooManyRedirectsException $e ) {
+            if ( defined( 'WP_CLI' ) ) {
+                \WP_CLI::warning( "Too many redirects from $url" );
```
This should probably be WsLog::l all the time, and just catch the exception without rethrowing it. It would be too surprising to have different crawl behavior between the CLI and the web UI.

bookwyrm replied:
That makes sense. Ideally the Too Many Redirects situation should be very uncommon with the redirect max increase from 1 to 2.

I think that we should start thinking about how to collect and surface crawl, process, and deploy errors and surface them in a report of some kind. Something that I struggled with a lot in the v6 code is that the git deploy would fail on a couple files and prevent the entire scrape from finishing.

bookwyrm committed:
Log redirect errors if encountered

Based on feedback from team, it makes more sense to simply log the error in the standard way and keep going with the crawl.
